src/components/AddExisitingStudentsModal.tsx

```diff
@@ -51,8 +51,12 @@ const AddExistingStudentsModal: React.FC<AddExistingStudentsModalProps> = ({
     let lowestOrderNumber = findLowestOrderNumber()
     selectedStudentsToAdd.forEach(student => {
       const newStudentInStudentGroupRef = studentsInStudentGroupsRef.doc()
-      const studentFacts = Object.values(student.studentFacts).filter(studentFact => {
-        return studentFact.value !== ''
+      const studentFacts: IStudentFact[] = []
+      Object.keys(student.studentFacts).forEach(key => {
```

Comment:
student facts are stored as keys to an object instead of objects in an array?

Reply:
On the student object in the backend studentFacts are stored as an object with each category as key and answer as value:
```
{
     "Favorite Food": "lasagna",
     "Favorite Beverage": "coke",
}
```
Then that object gets mapped over and stored as an array of objects on the studentInStudentGroup instance:

```
[
     {
      title: "Favorite Food",
      value: "lasagna"
     },
     {
      title: "Favorite Beverage",
      value: "coke"
     },
]
```
Then a random value can be spliced from that array, and once it's empty it gets refilled from the backend again.

Reply:
Why not store in the backend the way its used on the FE?

Reply:
Yeah, I was a little torn about what was better.  I ended up doing this because I also use that object to populate the inputs on the edit page with existing values.  Seemed cleaner to do lookups on an object rather than having to search an array for each value.

Reply:
Persuaded